components/cost-management/policies/update_namespace_labels.yaml

```
@@ -0,0 +1,72 @@
+---
+apiVersion: kyverno.io/v1
+kind: ClusterPolicy
+metadata:
+  name: propagate-cost-center-from-namespace
+  annotations:
+    policies.kyverno.io/title: Propagate Cost-Center from Namespace
+    policies.kyverno.io/category: Cost Management
+    policies.kyverno.io/severity: low
+    policies.kyverno.io/subject: Pod, Namespace
+    policies.kyverno.io/description: >-
+      This policy ensures that the cost-center label from a Namespace is propagated
+      to all Pods within that Namespace. It applies only to Namespaces with the
+      konflux-ci.dev/type: tenant label and ensures that the cost-center label exists
+      in the Namespace before applying the mutation.
+spec:
+  rules:
+    - name: propagate-missing-cost-center-from-namespace
+      match:
+        any:
+        - resources:
+            kinds:
+            - Pod
+            namespaceSelector:
+              matchLabels:
+                konflux-ci.dev/type: tenant
+      context:
+        - name: costcenterLabel
+          apiCall:
+            urlPath: "/api/v1/namespaces/{{ request.object.metadata.namespace }}"
+            method: GET
+            jmesPath: 'metadata.labels."cost-center" || ""'
+      preconditions:
+        all:
+          - key: "{{ costcenterLabel }}"
+            operator: NotEquals
+            value: ""
+      mutate:
+        targets:
+          - apiVersion: v1
+            kind: Pod
+        patchStrategicMerge:
+          metadata:
+            labels:
+              cost-center: "760"
+              insights_cost_management_optimizations: "true"
```

I think we should we add the Konflux domain to the label. WDYT?
```suggestion
              konflux-ci.dev/cost-center: "760"
              konflux-ci.dev/insights-cost-management-optimizations: "true"
```

---

For the cost management operator, the label insights-cost-management-optimizations is required in its current form, adding a domain prefix would prevent it from being recognized correctly. Additionally, in some cases, I believe the same cost-center label is applied from AWS. Including a domain would result in an extra label, introducing complexity in mapping.
Given these factors, I prefer to opt out of a domain-prefixed label for either of these cases would not be practical or beneficial.

---

thanks for the explanation